book/src/db/queries.rs

```diff
@@ -0,0 +1,49 @@
+use diesel::prelude::*;
+
+use super::schema::*;
+use super::{conn, models as db_models};
+use crate::rpc::models as rpc_models;
+
+pub fn get_book_by_id(book_id: i32) -> QueryResult<rpc_models::Book> {
```

Comment:
Here you are returning a rpc model, in the functions below you use db models. Where do you draw the line if a model belongs to the rpc or db models?

Reply:
`db_models` map directly to tables in the database (so they could implement the [`trait Identifiable`](https://docs.rs/diesel/1.4.5/diesel/associations/trait.Identifiable.html) which is helpful for associations and updates). All other models go into `rpc_models`.

Additionally the `rpc_models` are exposed to the api. And I plan to re-export all `db_models` which are returned by some rpc-function in the `rpc_models`.

Reply:
This sounds like it should be documented somewhere... 😬

Reply:
Yeah, I think so too.